### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: basic-numbers

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1034]
This issue describes how to implement the `basic-numbers` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This exercise should introduce the student to integer and floating point numbers in Common Lisp. Emphasis should be put on prefix notation, and all of the functions required for basic math and comparison should be covered.

## Learning objectives

- Know how to write integer and floating point literals
- Know how to perform basic math using `+`, `*`, `-`, `/`, `expt`, `log`, `sqrt`, and `mod`
- Be able to compare numbers using `=`, `/=`, `>`, `>=`, `<`, and `<=`
- Understand how prefix notation enables more than two numbers to be applied to one `+`, `*`, etc.
- Be aware of the special meanings of `-` and `/` when only one number is supplied

## Out of scope

- The different types of floating-point number
- Complex and rational numbers
- Any discussion of arbitrary precision arithmetic
- Random number generation
- The trigonometric functions
- Reader macros for numeric input (`#b0101`, `#c(2 2)`, etc)
- Parsing numbers from strings
- Converting numerical types
- Bitwise operations

## Concepts

- `integers`
- `floating-point-numbers`
- `arithmetic`

## Prerequisites

- `expressions`

## Resources to refer to

- [Common Lisp the Language](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node121.html)
- [Common Lisp Cookbook](https://lispcookbook.github.io/cl-cookbook/numbers.html)

### Hints

N/A

### After

Students could be referred to the [Common Lisp Hyperspec](http://clhs.lisp.se/Body/12_a.htm) to peruse the numerical functions available.

## Representer

- Currently, no extra logic needs to be added to the [representer](https://github.com/exercism/common-lisp-representer).

## Analyzer

- Currently, no extra logic needs to be added to the [analyzer](https://github.com/exercism/common-lisp-analyzer).

## Implementing

Aside from making a point of prefix notation, the functions should be purely mathematical (no conditions, list operations, or anything you couldn't put in a math equation). Try to pick some "interesting" equations that people will be familiar with, like the area of a circle or some other well-known equation.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by verdammelt** [comment]
I suggest for "floating point" that perhaps the differences between the different types of floating point numbers be left unsaid here or only mentioned in the `after.md`. Seems like we might want to have a whole concept about the type system... And I think for this exercise which appears intended to be relatively 'easy' we could just leave those details out. Thoughts?

**Comment by TheLostLambda** [comment]
I think that's a good point! I've added `"The different types of floating-point number"` to Out of Scope for now. 

As for an exercise regarding the type system, that might be a good idea, but we may need to see how a couple more exercises fit together before we are able to narrow down the scope of that one. Maybe would could have and exercise with things like `typep`, `typecase`, and `deftype`? That could also be the place to explain some of the hierarchy as well.
